### PR TITLE
Feature/repository dao

### PR DESCRIPTION
**Pull request by callicles**

### CODE REVIEW

File: app/models/services/RepositoryService.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+class RepositoryService @Inject() (repoDAO: RepositoryDAO) {
+
+  /**
+   * Saves or create a repository to the database according to the current needs
+   *
+   * @param name name of the repo to save or update
+   * @param addedLines number of added lines to the repo
+   * @param removedLines number of deleted lines from the repo
+   * @param score score of the repo
+   * @return the saved user
```
Comment (jacobtardieu) [inline app/models/services/RepositoryService.scala]: repo and not user.

File: app/models/services/RepositoryService.scala [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+   */
+  def save(name: String, addedLines: Option[Int], removedLines: Option[Int], score: Option[Int]): Future[Repository] = {
+    repoDAO.find(name).flatMap({
```
Comment (jacobtardieu) [inline app/models/services/RepositoryService.scala]: I don't think you need the `(` before the `{`. Also, why `flatMap` and not `map`?

File: app/models/services/RepositoryService.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          removedLines = removedLines.getOrElse(existingRepo.removedLines),
+          score = score.getOrElse(existingRepo.score)
+        ))
+      case None =>
+        repoDAO.save(Repository(
+          repoID = UUID.randomUUID(),
+          addedLines = addedLines.getOrElse(0),
+          removedLines = removedLines.getOrElse(0),
+          name = name,
+          score = 0
```
Comment (jacobtardieu) [inline app/models/services/RepositoryService.scala]: Why is the score hardcoded to `0` in that case?

File: app/models/services/UserServiceImpl.scala [hunk 4]
```diff
@@ -56,7 +56,8 @@ class UserServiceImpl @Inject() (userDAO: UserDAO) extends UserService {
           loginInfo = profile.loginInfo,
           fullName = profile.fullName,
           email = profile.email,
-          avatarURL = profile.avatarURL
+          avatarURL = profile.avatarURL,
+          karma = 0
```
Comment (jacobtardieu) [inline app/models/services/UserServiceImpl.scala]: Why `0`? Because it's a new user?

File: app/models/services/RepositoryService.scala (hunk 2 above)
Comment (callicles) [inline app/models/services/RepositoryService.scala]: With `map` the function won't return a `Future[Repository]` 
Error message: `Future[Repository] cannot conform to type _s`

File: app/models/services/UserServiceImpl.scala (hunk 4 above)
Comment (callicles) [inline app/models/services/UserServiceImpl.scala]: yes

Comment (jacobtardieu) [inline app/models/services/UserServiceImpl.scala]: And the karma of a new user is always `0`? Can't it be something else?

Comment (callicles) [inline app/models/services/UserServiceImpl.scala]: We could compute the Karma of a user at its creation, and set it as default but we don't have that functionality yet. Furthermore user creation is handled by silhouette so I don't think we can parametrize the function creation differently.

**Pull request merged**